### ISSUE TITLE
[Bug]:  "Reset controls" in addon panel pollutes SR announcements

### ISSUE THREAD

**Issue by leagrdv** [storybookjs/storybook#31436]
### Describe the bug

There are a few accessibility issues in the default ArgsTable rendering in Storybook that cause Axe and other a11y tools to raise warnings or errors. These issues can hinder usability for screen reader users and reduce compliance with accessibility standards such as WCAG.

### 1. Incorrect use of aria-readonly on radio button groups
Moved to https://github.com/storybookjs/storybook/issues/32250

### 2. Missing labels for select and input controls
Moved to https://github.com/storybookjs/storybook/issues/32250

### 3. Non-semantic clickable elements (`.sbdocs-expandable`)
<details>

<summary>Moved to #32246 </summary>
Elements with the class `.sbdocs-expandable` are interactive and reveal content on click, but they are implemented using div elements with no `role="button"` or keyboard accessibility.

These are not recognizable as interactive by assistive tech or keyboard users.

Recommendation: These elements should be implemented using `<button>` tags or given a `role="button"` with `tabindex="0"` and appropriate `aria-expanded` attributes, along with keyboard event handling for Enter/Space.
</details>

### 4. "Reset controls" in addon panel pollutes SR announcements

Table headings should not directly contain buttons or other secondary elements, because that causes the content of these buttons to be announced on every row visit in the entire table, ruining screenreader experience.

See https://adrianroselli.com/2025/07/check-uncheck-all-in-a-table.html for examples.

### Reproduction link

https://next.design-system.post.ch/

### Reproduction steps

Issue 1 (radio buttons): Go to `Components > Button` and the issue is visible on the first table of the page
Issue 2 (missing labels): Go to `Components > Button` and the issue is visible on the first table of the page
Issue 3 (expandable content): Go to `Components > Accordion` and the issue is visible on the second table of the page

### System

```bash
Storybook v8.4.7
```

**Comment by Sidnioulz** [comment]
Thanks for the report @leagrdv. I've found another issue that I'm adding to your issue description.

Work has started on https://github.com/storybookjs/storybook/pull/new/sidnioulz/issue-31436-table. Issue 4 should be fixed, the rest remains. That branch needs the IconButton API fix branch to be pushed before I can open a PR.

**Comment by Sidnioulz** [comment]
Rest continues on https://github.com/storybookjs/storybook/issues/32250
